### PR TITLE
[12.0][MIG] pos_picking_delayed

### PR DESCRIPTION
**Pull request by legalsylvain**
- Migrate pos_picking_delayed to V12.

This module postpones the creation of the picking of the pos.order, to make the creation of the pos.order faster. (~ two thirds of the time of the creation of the pos.order is used by the function create_picking())

- commit to review : https://github.com/OCA/pos/commit/c3a2eec6c436b29377ec1c7152220f2825f5e060
- original PR : https://github.com/OCA/pos/pull/293
- I refactored the module to use ``queue_job`` as suggested by @rousseldenis here https://github.com/OCA/pos/pull/293#discussion_r273411112

CC :  @chienandalu, @rafaelbn, @flotho 

Thanks for your review.

### CODE REVIEW

File: pos_picking_delayed/tests/test_module.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        self.assertEqual(
+            order.picking_id.id, False,
+            "Creating order via UI should not generate a picking if"
+            " feature is enabled")
+
+        # Test if a Queue Job has been generated
+        func_string = 'pos.order(%d,)._create_delayed_picking()' % (order.id)
+        queues = self.QueueJob.search([
+            ('func_string', '=', func_string)])
+        self.assertEqual(len(queues), 1, "Queue Job has not been created")
```
Comment (chienandalu) [inline pos_picking_delayed/tests/test_module.py:36]: Isn't it the opposite? I mean, "Queue Job has been created"

**Review by chienandalu — APPROVED** [review]
I finally could test the functionality in my local (queue_job doesn't work on runbot) and works great :smile: 

Thanks!

File: pos_picking_delayed/tests/test_module.py (hunk 1 above)
Comment (pedrobaeza) [reply pos_picking_delayed/tests/test_module.py:36]: No, this is the text to put when the condition is not met

Comment (chienandalu) [reply pos_picking_delayed/tests/test_module.py:36]: Sure. I said nothing :)

File: pos_picking_delayed/models/pos_order.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    @api.model
+    def _order_fields(self, ui_order):
+        res = super(PosOrder, self)._order_fields(ui_order)
+        res['has_picking_delayed'] = ui_order['has_picking_delayed']
+        return res
+
+    # Custom Section
+    @api.multi
+    @job
```
Comment (rousseldenis) [inline pos_picking_delayed/models/pos_order.py]: @legalsylvain Maybe a job channel here if you want user to customize its queues.

Comment (legalsylvain) [reply pos_picking_delayed/models/pos_order.py]: Hi @rousseldenis. Thanks for your review. Sorry, i'm not sure to understand. (Im' pretty newb in queue job...) 

Something like : 

``@job(default_channel='root.pos_picking_delayed')``

Comment (rousseldenis) [reply pos_picking_delayed/models/pos_order.py]: Yes! If it is not defined by user, it will take root by default.

**Comment by chienandalu** [comment]
@rousseldenis Your requests are done :)

**Review by rousseldenis — APPROVED** [review]

**Comment by chienandalu** [comment]
Thanks!

/ocabot merge

**Pull request closed without merging**